extensions/data-transfer/portability-data-transfer-rememberthemilk/src/main/java/org/dataportabilityproject/transfer/rememberthemilk/tasks/RememberTheMilkService.java

```diff
@@ -0,0 +1,146 @@
+  private RememberTheMilkSignatureGenerator signatureGenerator;
+
+  RememberTheMilkService(RememberTheMilkSignatureGenerator signatureGenerator) {
+    this.signatureGenerator = signatureGenerator;
+  }
+
+  public String createTimeline() throws IOException {
+    TimelineCreateResponse timelineCreateResponse =
+        makeRequest(
+            RememberTheMilkMethods.TIMELINES_CREATE,
+            ImmutableMap.of(),
+            TimelineCreateResponse.class);
+    checkState(!Strings.isNullOrEmpty(timelineCreateResponse.timeline));
+    return timelineCreateResponse.timeline;
+  }
+
+  public ListInfo createTaskList(String name, String timeline) throws IOException {
+    Map<String, String> params =
+        ImmutableMap.of("timeline", timeline, "name", ("Copy of: " + name));
+    ListAddResponse response =
+        makeRequest(RememberTheMilkMethods.LISTS_ADD, params, ListAddResponse.class);
+    checkState(response.listInfo != null, "Added list is null");
+    checkState(response.listInfo.id != 0, "Added list has id of zero");
+    return response.listInfo;
+  }
+
+  public TaskSeries createTask(String name, String timeline, String listId) throws IOException {
+    Map<String, String> params =
+        ImmutableMap.of("timeline", timeline, "name", name, "list_id", listId);
+    TaskAddResponse taskAddResponse =
+        makeRequest(RememberTheMilkMethods.TASK_ADD, params, TaskAddResponse.class);
+    return taskAddResponse.taskList.taskSeriesList.get(0);
+  }
+
+  public GetListResponse getList(String listId) throws IOException {
+    Map<String, String> params = ImmutableMap.of("list_id", listId);
+    return makeRequest(RememberTheMilkMethods.LISTS_GET_LIST, params, GetListResponse.class);
+  }
+
+  public GetListsResponse getLists() throws IOException{
+    return makeRequest(
+        RememberTheMilkMethods.LISTS_GET_LIST, ImmutableMap.of(), GetListsResponse.class);
+  }
+
+  private <T extends RememberTheMilkResponse> T makeRequest(
+      RememberTheMilkMethods method, Map<String, String> parameters, Class<T> dataClass)
+      throws IOException {
+
+    StringBuilder parameterString = new StringBuilder();
+    for (String key : parameters.keySet()) {
+      parameterString.append("&").append(key).append("=").append(parameters.get(key));
+    }
+
+    URL url = new URL(method.getUrl() + parameterString);
+    URL signedUrl = signatureGenerator.getSignature(url);
+
+    HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory();
+    HttpRequest getRequest = requestFactory.buildGetRequest(new GenericUrl(signedUrl));
+    getRequest.setParser(new XmlObjectParser(new XmlNamespaceDictionary().set("", "")));
+    HttpResponse response = getRequest.execute();
+    int statusCode = response.getStatusCode();
+    if (statusCode != 200) {
+      throw new IOException(
+          "Bad status code: " + statusCode + " error: " + response.getStatusMessage());
+    }
+
+    T parsedResponse = response.parseAs(dataClass);
```

Comment:
nit, this doesn't really matter but worth bringing up I guess. We're now using Jackson for all the other JSON parsing but I guess this is OK since (1) it's an extension, and they can pick their own libraries for doing things, (2) it (presumably) already works